### ISSUE TITLE
MPM Randomly Stops Mining to Selected Pool

### ISSUE THREAD

**Issue by Renegade5399** [MultiPoolMiner/MultiPoolMiner#901]
Example:

Have bat file set to mine to AHashPool. Will mine there fine for hours. Then at some point it drops AHashPool and beings to mine to all available pools in the Pools folder EXCEPT AHashPool.

Testing:

Created bat files to mine to other pools. Same result. Not sure of exact time before the symptom happens. Doesn't matter the pool. If set to mine to ZPool only, it will eventually stop mining there and mine to all other pools EXCEPT ZPool.

ExcludePoolName is not in use in the bat files.

Bat file example:

```
cd %~dp0

setx GPU_FORCE_64BIT_PTR 1
setx GPU_MAX_HEAP_SIZE 100
setx GPU_USE_SYNC_OBJECTS 1
setx GPU_MAX_ALLOC_PERCENT 100
setx GPU_SINGLE_ALLOC_PERCENT 100

set "command=& .\multipoolminer.ps1 -wallet 12345678 -username ABCDEFG -workername 1080Ti_Rig -region us -currency usd -type nvidia -poolname ahashpool -algorithm neoscrypt,blake2s,xevan,x17,sib,blakecoin,phi,c11,bitcore,hsr,yescrypt,skunk,nist5,lyra2v2,groestl,timetravel,tribus,scrypt,skein,myrgr,lbry,keccak,hmq1725 -donate 10 -watchdog -switchingprevention 2"

pwsh -noexit -executionpolicy bypass -windowstyle maximized -command "%command%"
powershell -version 5.0 -noexit -executionpolicy bypass -windowstyle maximized -command "%command%"
msiexec -i https://github.com/PowerShell/PowerShell/releases/download/v6.0.0-rc.2/PowerShell-6.0.0-rc.2-win-x64.msi -qb!
pwsh -noexit -executionpolicy bypass -windowstyle maximized -command "%command%"

pause
```
Not sure what is causing this. Happens on 2 different mining rigs.

**Comment by xfactor8383** [comment]
If you  want to mine just on Ahash, remove the other ps1 files of the other pools from the Pools folder. Just leave the Ahashpool.ps1 in there and that should take care of it. That's how I fixed my miner to just mine Hash Refinery.

**Comment by Renegade5399** [comment]
This does not fix this issues however. When only the AHashPool pool is in the folder, it starts mining, then says No pools available. It keeps mining whatever it was working on, but no longer profit switches.

**Comment by xfactor8383** [comment]
Ok so just to be clear it can't update via the API, but you do still mine on Ahash right?  If this is the case I've run into a similar issue mining on Hash Refinery...basically the site API can't be accessed when the timer ends, so none of the stats are updated and I stay mining on whatever I was mining. I generally think it's something to do with the pool website as I have a hard time accessing that status api sometimes just through the web. See if you can get this site to come up, if you can't consistently,it's a pool issue

**Comment by Renegade5399** [comment]
I thought this too! It's why I tested it with all the pools (minus the 2 new ones). After a bit, it will just start mining to all the other pools EXCEPT the one I want. If I remove all the pools from the pools folder except the one I want, it mines that pool fine for a bit, then starts saying No pools available. It keeps mining the coin it was on though at least. This behavior is the same for all pools though, thus why I finally opened the issue here as I think it may be a MPM issue.

**Comment by xfactor8383** [comment]
Ok well that makes me nervous I might haven't he same issue. I haven;t done any testing.. and am happy to mine just Neoscrypt for now.  For me it seems to have just happened in the last two days... I am still running 2.7.01. Let me know if you figure it out.

**Comment by fonyo** [comment]
Seems to be the pools going down, not responding and automatic failover kicks in. Nicehash is having a lot of trouble, too. MPH is stable.

**Comment by xfactor8383** [comment]
Hey- Ok so I had this problem real bad this am. I couldn't connect to any pools ( Hashrefinery, ZPool, Nicehash), because I got the API ( pool) not found error. I noticed that even running the invoke-RestMethod in power-shell with a pools api created a long delay before the api status was actually pulled up. To solve it I changed the timer in the (pool).ps1 file from 10 second to 60 seconds.  60 seconds is probably overkill  but it's working for me now. 

$HashRefinery_Request = Invoke-RestMethod "http://pool.hashrefinery.com/api/status" -UseBasicParsing -TimeoutSec 60 -ErrorAction Stop
